test/exporter_test.rb

```diff
@@ -0,0 +1,100 @@
+require_relative 'test_helper'
+module Dynflow
+  module ExporterTest
+    describe ::Dynflow::Exporters::Hash do
+
+      before do 
+        @world = WorldFactory.create_world
+      end
+
+      let :execution_plan do
+        @world.plan(Support::CodeWorkflowExample::FastCommit, 'sha' => 'abc123')
+      end
+
+      let :ref_action do
+        @world.persistence.load_action(execution_plan.steps[3]) 
+      end
+
+      describe 'execution plan export' do
+
+        before do
+          execution_plan.save
+          @exporter = Exporters::Hash.new(execution_plan)
+        end
+
+        it 'exports execution plan items properly' do
+          direct_keys = [:id, :state, :result, :started_at, :ended_at, :execution_time, :real_time]
+          changed_keys = [:execution_history, :phase, :sub_plans, :delay_record]
+          all_keys = direct_keys + changed_keys
+          exported_plan = @exporter.export
+          execution_plan_hash = execution_plan.to_hash
+          assert exported_plan.keys.all? { |key| all_keys.include? key }
+          direct_keys.each do |key|
+            exported_plan[key].must_equal execution_plan_hash[key]
+          end
+          exported_plan[:execution_history].each do |history|
+            history[:time].must_be Time
+          end
+          exported_plan[:sub_plans].must_be_empty
+          exported_plan[:delay_record].must_be_empty
+        end
+
+        it 'exports finalize phase properly' do
+          exported_plan = @exporter.export
+          finalize = exported_plan[:phase][:finalize]
+          finalize[:type].must_equal 'sequence'
+          finalize[:steps].count.must_equal execution_plan.finalize_flow.flows.count
+        end
+
+      end
+    end
+
+    describe ::Dynflow::Exporters::Tar do
```

Comment:
Do you plan on writing JSON and HTML exporter tests as well? I would be ok with some more high level tests, with more end-2-end tests rather than the stubbing once, unless the time penalty was too high.